Sources/Core/Microsoft.StreamProcessing.Provider/Extensions/Linq.cs

```diff
@@ -0,0 +1,76 @@
+﻿// *********************************************************************
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License
+// *********************************************************************
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace Microsoft.StreamProcessing.Provider
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public struct Empty
+    { }
+
+    /// <summary>
+    /// The extension methods over interface IQStreamable
+    /// </summary>
+    public static partial class QStreamableStatic
+    {
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="TLeft"></typeparam>
+        /// <typeparam name="TRight"></typeparam>
+        /// <typeparam name="TResult"></typeparam>
+        /// <param name="left"></param>
+        /// <param name="right"></param>
+        /// <param name="resultSelector"></param>
+        /// <returns></returns>
+        public static IQStreamable<TResult> SelectMany<TLeft, TRight, TResult>(
+            this IQStreamable<TLeft> left,
+            Func<Empty, IQStreamable<TRight>> right,
+            Expression<Func<TLeft, TRight, TResult>> resultSelector)
+            => Join(left, right(default), o => true, o => true, resultSelector);
```

Comment:
Why not just have "right" be a Func<IQStreamable<TRight>> or even just IQStreamable<TRight>?

Reply:
This is a requirement of the LINQ syntactic sugar.

Reply:
To be more specific:

var foo = from s1 in streamable1 from s2 in streamable2...

compiles down to this and requires a func with a parameter.